File: problems/1897-redistribute-characters-to-make-all-strings-equal/solution.py
class Solution:
    def makeEqual(self, words: List[str]) -> bool:
        counts = {}
        total_length = 0
        for word in words:
            total_length += len(word)
            for char in word:
                counts[char] = counts.get(char, 0) + 1
        for count in counts.values():
            if count % len(words) != 0:
                return False
        return True

File: problems/1897-redistribute-characters-to-make-all-strings-equal/solution.cpp
class Solution {
public:
    bool makeEqual(vector<string>& words) {
        vector<int> count(26, 0);
        for (const string& word : words) {
            for (char c : word) {
                count[c - 'a']++;
            }
        }
        for (int c : count) {
            if (c % words.size() != 0) {
                return false;
            }
        }
        return true;
    }
};